# CAT

This image implements:
* [CAT v5.3](https://github.com/dutilh/CAT)

It can be accessed at [docker hub](https://hub.docker.com/u/tpaisie).

Contig Annotation Tool (CAT) and Bin Annotation Tool (BAT) are pipelines for the taxonomic classification of long DNA sequences and metagenome assembled genomes (MAGs / bins) of both known and (highly) unknown microorganisms, as generated by contemporary metagenomics studies. The core algorithm of both programs involves gene calling, mapping of predicted ORFs against a protein database, and voting-based classification of the entire contig / MAG based on classification of the individual ORFs. CAT and BAT can be run from intermediate steps if files are formated appropriately.

## WARNING: CAT needs a large database to run 

## Example analysis
Get test data:
```
# Download test data
wget -nv https://raw.githubusercontent.com/taylorpaisie/docker_containers/main/rdp/2.14/16S_rRNA_gene.Burkholderia_pseudomallei.2002721184.AY305776.1.fasta -O 16S_test.fa
wget -nv https://raw.githubusercontent.com/taylorpaisie/docker_containers/main/rdp/2.14/18S_rRNA_gene.Homo_sapiens.T2T-CHM13v2.0.Chromosome13.fasta -O 18S_test.fa
```

Use RDP Classifier to get taxonomic assignments for bacterial and archaeal 16S rRNA sequences:
```
classifier classify -o taxa_16S_test.txt 16S_test.fa
classifier classify -o taxa_18S_test.txt 18S_test.fa
```

## Output
```
head -2 taxa_16S_test.txt

AY305776.1		Root	rootrank	1.0	Bacteria	domain	1.0	Pseudomonadota	phylum	1.0	Betaproteobacteria	class	1.0	Burkholderiales	order	1.0	Burkholderiaceae	family	1.0	Burkholderia	genus	1.0
```